src/dos/drive_local.cpp

```diff
@@ -46,32 +47,53 @@
 #include "cross.h"
 #include "inout.h"
 
-bool localDrive::FileCreate(DOS_File * * file,char * name,uint16_t /*attributes*/) {
-//TODO Maybe care for attributes but not likely
+bool localDrive::FileCreate(DOS_File * * file,char * name,uint16_t attributes) {
+	// Don't allow overwriting read-only files.
+	uint16_t test_attr = 0;
+	if (GetFileAttr(name, &test_attr) && (test_attr & DOS_ATTR_READ_ONLY)) {
+		DOS_SetError(DOSERR_ACCESS_DENIED);
+		return false;
+	}
+
+	bool existing_file = FileExists(name);
+
 	char newname[CROSS_LEN];
 	safe_strcpy(newname, basedir);
 	safe_strcat(newname, name);
 	CROSS_FILENAME(newname);
-	char* temp_name = dirCache.GetExpandName(newname); //Can only be used in till a new drive_cache action is preformed */
-	/* Test if file exists (so we need to truncate it). don't add to dirCache then */
-	bool existing_file = false;
-	
-	FILE * test = fopen_wrap(temp_name,"rb+");
-	if (test) {
-		fclose(test);
-		existing_file=true;
-
+	FILE* fp = nullptr;
+
+#ifdef WIN32
+	DWORD win_attribs = attributes ? (DWORD)attributes : FILE_ATTRIBUTE_NORMAL;
+	HANDLE win_fhandle = CreateFile(newname,
+	                                GENERIC_READ | GENERIC_WRITE,
+	                                0,
+	                                nullptr,
+	                                CREATE_ALWAYS,
+	                                win_attribs,
+	                                nullptr);
+	if (win_fhandle != INVALID_HANDLE_VALUE) {
+		int fd = _open_osfhandle((intptr_t)win_fhandle, _O_RDWR | _O_BINARY);
+		fp = _fdopen(fd, "wb+");
 	}
-	
-	FILE * hand = fopen_wrap(temp_name,"wb+");
-	if (!hand) {
+#else
+	constexpr int rw_perms = S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH;
```

Comment:
0644 is the default for new files on Linux.  So no write permissions for group or other.  I would leave it at that unless you have a good reason not to.

Reply:
Actually, when you create a file on Linux, the process's current umask is subtracted from permission bits and umask is usually 022 by default. So it's perfectly fine to pass 0666.
https://man7.org/linux/man-pages/man2/umask.2.html

Reply:
I stand corrected.  This is fine then.